packages/eslint-plugin/src/rules/index.ts

```diff
@@ -184,7 +184,6 @@ export default {
   'no-non-null-asserted-nullish-coalescing': noNonNullAssertedNullishCoalescing,
   'no-non-null-asserted-optional-chain': noNonNullAssertedOptionalChain,
   'no-non-null-assertion': noNonNullAssertion,
-  'no-parameter-properties': noParameterProperties,
```

Comment:
@bradzacher is there any precedent in this repo you prefer for renaming a rule?

I'm thinking about writing a standard helper that wraps a rule and logs that it's been renamed.

```ts
export const noParameterProperties = wrapRenamedRule(parameterProperties, 'no-parameter-properties');
```

```ts
'no-parameter-properties': noParameterProperties,
```

Reply:
Logging isn't a good idea because it breaks tooling that expects JSON output and such from ESLint runs.

If you want - you can merge this now as a new rule and mark the old rule as deprecated.
then in the next major we can remove the new rule entirely.
The deprecated metadata will cause ESLint to add something to the lint result... it's invisible but it's there.

IIRC deprecated rules won't be shown in our website either.

Reply:
Thanks, makes sense. I filed #4736 since https://typescript-eslint.io/rules/no-implicit-any-catch/ is still showing up.

Reply:
Re-requesting review as this is my interpretation of the rename/deprecation strategy.

Reply:
...or, I will do the closest equivalent the GitHub UI will allow me.

I'll also use `replacedBy` per https://github.com/typescript-eslint/typescript-eslint/issues/4736#issuecomment-1079734810.